pennylane/ops/qubit/qchem_ops.py

```diff
@@ -223,13 +223,41 @@ def compute_decomposition(phi, wires):
         **Example:**
 
         >>> qml.SingleExcitation.compute_decomposition(1.23, wires=(0,1))
-        [CNOT(wires=[0, 1]), CRY(1.23, wires=[1, 0]), CNOT(wires=[0, 1])]
+        [Adjoint(T(wires=[0])),
+         Hadamard(wires=[0]),
+         S(wires=[0]),
+         Adjoint(T(wires=[1])),
+         Adjoint(S(wires=[1])),
+         Hadamard(wires=[1]),
+         CNOT(wires=[1, 0]),
+         RZ(-0.615, wires=[0]),
+         RY(0.615, wires=[1]),
+         CNOT(wires=[1, 0]),
+         Adjoint(S(wires=[0])),
+         Hadamard(wires=[0]),
+         T(wires=[0]),
+         Hadamard(wires=[1]),
+         S(wires=[1]),
+         T(wires=[1])]
 
         """
         decomp_ops = [
```

Comment:
Can `SingleExcitationPlus` and `SingleExcitationMinus` gates benefit from this new decomposition as well?

Reply:
Yes this is likely! I can try to work that out as well. Should I make a separate PR or would it be fine to include in this one?

Reply:
I think it should be fine to include it in this one.

Reply:
Okay I've done some digging. In the original decomposition, we have

```python
decomp_ops = [
    qml.PauliX(wires=wires[0]),
    qml.PauliX(wires=wires[1]),
    qml.ControlledPhaseShift(-phi / 2, wires=[wires[1], wires[0]]),
    qml.PauliX(wires=wires[0]),
    qml.PauliX(wires=wires[1]),
    qml.ControlledPhaseShift(-phi / 2, wires=[wires[0], wires[1]]),

    qml.CNOT(wires=[wires[0], wires[1]]),
    qml.CRY(phi, wires=[wires[1], wires[0]]),
    qml.CNOT(wires=[wires[0], wires[1]]),
]
```
The first part provides the phases on the first/last diagonal elements, and the second part is just a regular single excitation. 

As-is, that's 8 CNOTs. We could swap out the single excitation part alone to get it down to 6. It can in principle be implemented using only 3, but the decomposition yields a global phase that depends on the rotation angle. There is some work required to get it to match (which we need to do so that everything will work okay when the operation is controlled on some other qubit), it requires rejigging a lot of the rotations in terms of RZ and Phaseshift, and ends up pretty ugly. Are the `Plus` and `Minus` versions sufficiently common to use in algorithms (that would run on hardware) that would warrant doing this?

Here is the one for `SingleExcitationPlus` for reference:

```python
decomp_ops = [
    qml.S(wires=0)
    qml.T(wires=0)
    qml.Hadamard(wires=0)
    qml.adjoint(qml.S)(wires=0)
    qml.adjoint(qml.T)(wires=0)

    qml.RZ(np.pi/4, wires=1)
    qml.Hadamard(wires=1)
    qml.RZ(np.pi/4, wires=1)

    qml.CNOT(wires=[1, 0])
    qml.PhaseShift(np.pi/2 - phi/2, wires=0)
    qml.RY(np.pi/2 - phi/2, wires=1)
    qml.CNOT(wires=[0, 1])
    qml.RY(np.pi/2 - phi/2, wires=1)
    qml.CNOT(wires=[1, 0])

    qml.adjoint(qml.T)(wires=0)
    qml.Hadamard(wires=0)
    qml.T(wires=0)

    qml.S(wires=1)
    qml.RZ(np.pi/4, wires=1)
    qml.Hadamard(wires=1)
    qml.RZ(np.pi/4, wires=1)
    qml.S(wires=1)
]
```

Reply:
I don't think these gates are used that commonly on the hardware. So we can let them be for now (or just grab the straightforward two CNOTs decrease via single excitation part).

Reply:
Sounds good, if they are not used much, let's just leave them be. Plus if I recall correctly they were originally used _for_ `SingleExcitation` in some capacity (either its decomposition or for the gradient rule, can't remember exactly).